src/HashedExpression/Internal/Normalize.hs

```diff
@@ -499,8 +558,12 @@ expandPiecewiseRealImag exp@(mp, n)
       | RealImag re im <- retrieveNode nId mp = Just (re, im)
       | otherwise = Nothing
 
--- | piecewise marks condition [branch1, branch2, ..] -->
--- branch1 * piecewise marks condition [1, 0, 0, ..] + branch2 * piecewise marks condition [0, 1, 0, ..] + ..
+-- FIXME: This one needs a better explanation.
+
```

Comment:
This rules is for breaking down piecewise expressions.

E.g:

if x < 1 then x + 1 else x + y = (x + 1) * (if x < 1 then 1 else 0) + (x + y) * (if x < 1 then 0 else 1)